contacts/main.js

```diff
@@ -0,0 +1,563 @@
+// the main list of contacts- person objects
+let contactsList;
+
+// person entity
+class Person {
+  constructor(person) {
+    this.id = person.id;
+    this.firstName = person.firstName;
+    this.lastName = person.lastName;
+    this.image = person.image;
+    this.likes = person.likes;
+  }
+  getFullNameAndLikes() {
+    return `${this.id} ${this.firstName} ${this.lastName} ${this.likes}`;
+  }
+}
+/////////////////////////////////////////////////////////////////////////
+function getInitialListOfContacts() {
+  let array = [];
+  var id;
+  var obj;
+
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Alin',
+    lastName: 'Ionescu',
+    image: 'images/profile-pics/pic1.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Leona',
+    lastName: 'Sherman',
+    image: 'images/profile-pics/pic2.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Annette',
+    lastName: 'Carlson',
+    image: 'images/profile-pics/pic3.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Ervin',
+    lastName: 'Watkins',
+    image: 'images/profile-pics/pic4.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Philip',
+    lastName: 'Pope',
+    image: 'images/profile-pics/pic5.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Viorel',
+    lastName: 'Popescu',
+    image: 'images/profile-pics/pic6.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Elon',
+    lastName: 'Musk',
+    image: 'images/profile-pics/pic7.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  let retunrArray = [];
+  array.forEach((item) => {
+    let person = new Person(item);
+    retunrArray.push(person);
+  })
+  return retunrArray;
+}
+
+function saveDataToLocalStorage() {
+  localStorage.setItem('contacts', JSON.stringify(contactsList));
+}
+
+function getContactsFromLocalStorage() {
+  return JSON.parse(localStorage.getItem('contacts'));
+}
+
+function getContactsAccordingly() {
+  let array = getContactsFromLocalStorage();
+  // if we have nothing in the local storage we get the initial list
+  if (array === null || array.length === 0)
+    array = getInitialListOfContacts()
+  return array;
+}
+
+/////////////////////////////////////////////////////////////////////////
+
+function getNumberOfNonZeroLikes() {
```

Comment:
This function seems to be doing more that is needed. It gets the number of non zero likes, but when you use it, you only check if it's different from 0. So I suggest rewritting this function a little bit. Name it something like 'ratedContactsExist' and make it return true whenever it finds the first contact with available rating, and false otherwise.